src/skagent/ann.py

```diff
@@ -657,51 +685,44 @@ def train_block_value_and_policy_nn(
         The value network to train
     inputs : Grid
         Input grid containing states and shocks
-    policy_loss_function : callable
-        Loss function for policy training (takes decision_function, input_grid)
-    value_loss_function : callable
-        Loss function for value training (takes value_function, input_grid)
+    loss_function : callable
+        Unified loss function that takes (value_function, decision_function, input_grid)
+        and trains both networks simultaneously
```

Comment:
Is the Euler residual loss going to be like this?

If this is tightly coupled to the Bellman loss function, the abstractions are strange.